src/days-2-4-adv/adv.py

```diff
@@ -32,15 +43,54 @@
 room['narrow'].w_to = room['foyer']
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
-
 #
 # Main
 #
 
 # Make a new player object that is currently in the 'outside' room.
+p=Player(input('\nWhat is your name?'),room['outside'])
 
 # Write a loop that:
-#
+
+direction = "\nWhat direction would you like to walk in?[n,s,e,w or q to quit]\nYou could also enter commands like take sword or get coins to grab the availabe items in the room"
+
+print(f'\nYour current location is:{p.currentRoom.name}.')
+print('\nItems availabe in this room are:\n') 
+p.currentRoom.showItems()
+print(f'{p.currentRoom.text}')
+print(direction)
+
+
+valid_directions = {"n": "n", "s": "s", "e": "e", "w": "w","f": "n", "b": "s", "r": "e", "l": "w"}
+
+while True:        
+        cmds = input("-> ").lower().split(' ')
+        #print(cmds)
+        cmd=cmds[0]
+        #print(cmd)
+        if cmd == "q":
+                print('You choose to quit.')
+                break
+        elif cmd in valid_directions:
+                p.travel(valid_directions[cmd])
+                print(f'Items available in this room are:')
+                p.currentRoom.showItems()
+        elif cmd == "take" or cmd == "get":
+                result=p.currentRoom.getItem(cmds[1])
+                #print(f'{result}')
+                if result!=None:
+                        p.addItem(result)
+                        room[p.currentRoom.name].removeItem(result)
```

Comment:
I just had a thought - maybe it is not such a good idea to have coins in every room??? I have no idea if this is the cause of the problem, but it might be an issue. You could have different kinds of treasure - jewels, gold, etc. just to be on the safe side.